packages/rs-drive-abci/src/execution/types/validation_operation/signature_verification_operation.rs

```diff
@@ -0,0 +1,110 @@
+use crate::error::Error;
+use crate::execution::types::validation_operation::{OperationLike, ValidationOperation};
+use dpp::fee::Credits;
+use dpp::identity::identity_public_key::accessors::v0::IdentityPublicKeyGettersV0;
+use dpp::identity::state_transition::AssetLockProved;
+use dpp::identity::{KeyType, PartialIdentity};
+use dpp::prelude::AssetLockProof;
+use dpp::state_transition::identity_create_transition::accessors::IdentityCreateTransitionAccessorsV0;
+use dpp::state_transition::identity_update_transition::accessors::IdentityUpdateTransitionAccessorsV0;
+use dpp::state_transition::public_key_in_creation::accessors::IdentityPublicKeyInCreationV0Getters;
+use dpp::state_transition::{StateTransition, StateTransitionIdentitySigned};
+use dpp::version::PlatformVersion;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SignatureVerificationOperation {
+    pub signature_type: KeyType,
+}
+
+impl SignatureVerificationOperation {
+    pub fn new(signature_type: KeyType) -> Self {
+        Self { signature_type }
+    }
+}
+
+impl OperationLike for SignatureVerificationOperation {
+    fn processing_cost(&self, platform_version: &PlatformVersion) -> Result<Credits, Error> {
+        Ok(self
+            .signature_type
+            .signature_verify_cost(platform_version)?)
+    }
+
+    fn storage_cost(&self, _platform_version: &PlatformVersion) -> Result<Credits, Error> {
+        Ok(0)
+    }
+}
+
+pub fn signature_verification_operations_from_state_transition(
+    state_transition: &StateTransition,
+    identity: Option<&PartialIdentity>,
+) -> Vec<ValidationOperation> {
```

Comment:
I believe the better approach is to get the operations from the execution context.

Reply:
Also this was not versioned